TestingHelpers/MockDirectoryInfo.cs

```diff
@@ -284,5 +329,10 @@ public override DirectoryInfoBase Root
                 return new MockDirectoryInfo(mockFileDataAccessor, mockFileDataAccessor.Directory.GetDirectoryRoot(FullName));
             }
         }
+
+        private void CheckDirectoryExists() 
+        {
+            if (MockFileData == null) throw new FileNotFoundException("File not found", directoryPath);
```

Comment:
This behaviour isn't consistent with what I see the real filesystem do on my machine.

Local test:

```
PS C:\Users\tatha> $a = [System.IO.DirectoryInfo]::new("c:\temperererer")
PS C:\Users\tatha> $a.LastWriteTime

Monday, 1 January 1601 11:00:00


PS C:\Users\tatha> $a.Exists
False
```

Note that there is no exception thrown.